addons/stock/models/stock_production_lot.py

```diff
@@ -65,6 +65,14 @@ def _compute_display_complete(self):
     @api.model_create_multi
     def create(self, vals_list):
         self._check_create()
+        for vals in vals_list:
+            if vals.get('name'):
+                name, product_id, company_id = vals['name'], vals['product_id'], vals['company_id']
+                lot = self.search([('name', '=', name),
+                                   ('product_id', '=', product_id),
+                                   ('company_id', '=', company_id)])
+                if lot:
+                    raise UserError(_('The combination of serial number and product must be unique across a company.\nThe serial number "%s" for product "%s" has already been assigned.' % (lot.name, lot.product_id.display_name)))
```

Comment:
doesn't look good to perform a search at each create? can you explain why you need this change?

Reply:
Because in stock.production.lot, we use sql constraint to prevent the duplicates. I can't customize the error message in sql constraints.

Reply:
the sql constraint is most probably there because it is faster than a python one, now here you defeats all the speed gains by doing the work of the constraint a second time at create just for the sake of an error message...

i see multiple solutions here
- cancel this part of the task
- change the sql constraint to a python one with a proper message **if** we're sure it doesn't harm the performance too bad and for that you'll need to give me the delays to process the receipt and delivery for a large number of products (let's say 2000 serials)

Reply:
It takes 14 sec to finish 2000 creations and with previous code it's 10 sec. Quite a harm to the performance. I think we should cancel this part.

Reply:
can you try by converting this constaint to a python one?